presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -115,7 +116,8 @@ public HiveSplitManager(
                 hiveClientConfig.getMaxPartitionBatchSize(),
                 hiveClientConfig.getMaxInitialSplits(),
                 hiveClientConfig.getSplitLoaderConcurrency(),
-                hiveClientConfig.getRecursiveDirWalkerEnabled());
+                hiveClientConfig.getRecursiveDirWalkerEnabled(),
+                new S3SelectPushdown(hiveClientConfig));
```

Comment:
As a general coding convention, we never store config objects in classes, since config objects are mutable beans. Instead, the constructor should extract the values into immutable fields.

Reply:
Fixed